File: src/Python/core/jumps.py
from __future__ import annotations

class JumpSignal(Exception):
    def __init__(self, label: str, delay_ms: int = 0):
        if not label or not str(label).strip():
            raise ValueError("label must be non-empty")
        super().__init__(f"jump to '{label}' after {int(delay_ms)}ms")
        self.label = str(label)
        self.delay_ms = max(0, int(delay_ms))

def jump_now(label: str) -> None:
    raise JumpSignal(label, 0)

def jump_after(label: str, delay_ms: int) -> None:
    raise JumpSignal(label, delay_ms)

File: src/typescript/core/jumps.ts
export class JumpSignal extends Error {
  public readonly label: string;
  public readonly delayMillis: number;

  constructor(label: string, delayMillis: number = 0) {
    if (!label || String(label).trim().length === 0) {
      throw new Error("label must be non-empty");
    }
    super(`jump to '${label}' after ${Number(delayMillis)}ms`);
    this.name = "JumpSignal";
    this.label = String(label);
    this.delayMillis = Math.max(0, Number(delayMillis));
  }
}

export function jumpNow(label: string): never {
  throw new JumpSignal(label, 0);
}

export function jumpAfter(label: string, delayMillis: number): never {
  throw new JumpSignal(label, delayMillis);
}
